Mathlib/AlgebraicTopology/SimplexCategory/GeneratorsRelations/EpiMono.lean

```diff
@@ -0,0 +1,354 @@
+/-
+Copyright (c) 2025 Robin Carlier. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Robin Carlier
+-/
+import Mathlib.AlgebraicTopology.SimplexCategory.GeneratorsRelations.Basic
+/-! # Epi-mono factorisation in the simplex category presented by generators and relations
+
+This file aims to establish that there is a nice epi-mono factorisation in `SimplexCategoryGenRel`.
+More precisely, we introduce two (inductively-defined) morphism property `P_δ` and `P_σ` that
+single out morphisms that are compositions of `δ i` (resp. `σ i`).
+
+The main result of this file is `exists_P_σ_P_δ_factorisation`, which asserts that every
+moprhism as a decomposition of a `P_σ` followed by a `P_δ`.
+
+-/
+
+namespace SimplexCategoryGenRel
+open CategoryTheory
+
+section EpiMono
+
+/-- `δ i` is a split monomorphism thanks to the simplicial identities. -/
+def SplitMonoδ {n : ℕ} {i : Fin (n + 2)} : SplitMono (δ i) where
+  retraction := by
+    induction i using Fin.lastCases with
+    | last => exact σ n
+    | cast i => exact σ i
+  id := by
+    cases i using Fin.lastCases
+    · simp only [Fin.natCast_eq_last, Fin.lastCases_last]
+      exact δ_comp_σ_succ
+    · simp only [Fin.natCast_eq_last, Fin.lastCases_castSucc]
+      exact δ_comp_σ_self
+
+instance {n : ℕ} {i : Fin (n + 2)} : IsSplitMono (δ i) := .mk' SplitMonoδ
+
+/-- `δ i` is a split epimorphism thanks to the simplicial identities. -/
+def SplitEpiσ {n : ℕ} {i : Fin (n + 1)} : SplitEpi (σ i) where
+  section_ := δ i.castSucc
+  id := δ_comp_σ_self
+
+instance {n : ℕ} {i : Fin (n + 1)} : IsSplitEpi (σ i) := .mk' SplitEpiσ
+
+/-- Auxiliary predicate to express that a morphism is purely a composition of `σ i`s. -/
+inductive P_σ : MorphismProperty SimplexCategoryGenRel
```

Comment:
It could be nice to generalize this to the multiplicative morphism property generated by `W : MorphismProperty C` (using only identifies and precomposition, for example), and obtain two general induction principles (one relying on precomposition, and the other on postcomposition).